pydeseq2/dds.py

```diff
@@ -1016,6 +1037,34 @@ def cooks_outlier(self):
         self.varm["_pvalue_cooks_outlier"] = cooks_outlier
         return self.varm["_pvalue_cooks_outlier"]
 
+    def to_picklable_anndata(self) -> ad.AnnData:
+        """Convert the DESeqDataSet to a picklable AnnData object.
+
+        Builds an AnnData object from the DESeqDataSet with the same data, but converts
+        the design matrix to a DataFrame to remove the formulaic model_spec attribute,
+        which is not picklable.
+
+        Returns
+        -------
+        anndata.AnnData
+            The AnnData object, without DeseqDataSet unpicklable attributes.
+        """
+        # Initialize an AnnData object
+        adata = ad.AnnData(
+            X=self.X,
+            obs=self.obs,
+            var=self.var,
+            obsm=self.obsm,
+            varm=self.varm,
+            uns=self.uns,
+            layers=self.layers,
+        )
+
+        # Convert the design matrix to a DataFrame to remove model_spec
+        adata.obsm["design_matrix"] = pd.DataFrame(adata.obsm["design_matrix"])
+
+        return adata
+
```

Comment:
As mentioned on zulip, I think a preferred way would be to implement a 
`.save()` and (static) `.load()` method that saves the anndata as h5ad and the state of the model as pickle or any other suited format. 

The `.load()` should be able to restore the full state of the dds object, including the formulaic metadata ideally.

Reply:
I think that's a good suggestion, but it should probably be treated in a different PR to avoid bloating this one (which is already pretty big).

Also as discussed on Zulip, I think that inheriting from the `AnnData` class wasn't the best choice, and that `DeseqDataSet` should have an `AnnData` component instead. But again, another PR :)

Reply:
Opened #330 and #331 to keep track of this.